cliapi/providers/azure.py

```diff
@@ -0,0 +1,66 @@
+from cliapi.cliapi_lib import Provider, cliapi_compile
+from cliapi.what_cloud import determine_provider
+
+import json
+import os
+import uuid
+import urllib.request, urllib.error, urllib.parse
+import xml.etree.ElementTree as ET
+
+try:
+    # ----- raises an error on import if not running in a "proper" cloud...
+    cp = determine_provider()
+    if cp != 'azure':
+        raise Exception("not a valid provider")
+except Exception as e:
+    raise e
+
+provider = Provider()
+
+scoops = {
+    'instance-name': "['meta_data']['compute']['name']",
+    'mac': "['meta_data']['network']['interface'][0]['macAddress']",
+    'location': "['meta_data']['compute']['location']",
+    'external-ip': "['meta_data']['network']['interface'][0]"
+                   "['ipv4']['ipAddress'][0]['publicIpAddress']",
+    'internal-ip': "['meta_data']['network']['interface'][0]"
+                   "['ipv4']['ipAddress'][0]['privateIpAddress']",
+    'cloud-service': "['get_cloud_service']"
+}
+
```

Comment:
scoops should not be forced to be hard coded. For example the scoop between API version 1 and 2 is going to be different for cloud metadata querying. This would imply we either have to hard code the scoop for every API version and monitor the framework provider and add a new scoop when a new API version is introduced or we have to generate the scoop dynamically based on the API version (there should be a default) provided to the tool on the command line.